src/frontend/komponenter/Fagsak/Vedtak/VedtakBegrunnelserTabell/VedtaksperioderMedBegrunnelser/TrekkILøpendeUtbetaling/TrekkILøpendeUtbetalingListe.tsx

```diff
@@ -0,0 +1,112 @@
+}> = ({ visTrekkILøpendeUtbetalinger, åpenBehandling }) => {
+    const { request } = useHttp();
+    const [trekkILøpendeUtbetalinger, settTrekkILøpendeUtbetalinger] = useState<
+        ITrekkILøpendeUtbetaling[]
+    >([]);
+
+    const hentTrekkILøpendeUtbetalinger = async () => {
+        const trekk = await request<void, IRestTrekkILøpendeUtbetaling[]>({
+            method: 'GET',
+            url: '/familie-ba-sak/api/trekk-i-loepende-utbetaling',
+        });
+        if (trekk.status !== RessursStatus.SUKSESS) {
+            settTrekkILøpendeUtbetalinger([]);
+        }
+        if (trekk.status === RessursStatus.SUKSESS) {
+            const data = trekk.data;
+            const transformert = data.map(d => fraRest(d));
+            settTrekkILøpendeUtbetalinger(transformert);
+            return trekk;
+        }
+    };
+
+    const leggTilNyPeriode = () => {
+        settTrekkILøpendeUtbetalinger([
+            ...trekkILøpendeUtbetalinger,
+            {
+                id: 0,
+                behandlingId: åpenBehandling.behandlingId,
+                feilutbetaltBeløp: 0,
+                periode: {},
+            },
+        ]);
+    };
+
+    function fraRest(
+        trekkILøpendeUtbetaling: IRestTrekkILøpendeUtbetaling
+    ): ITrekkILøpendeUtbetaling {
+        return {
+            ...trekkILøpendeUtbetaling,
+            id: trekkILøpendeUtbetaling.identifikator.id,
+            behandlingId: trekkILøpendeUtbetaling.identifikator.behandlingId,
+            periode: {
+                fom: trekkILøpendeUtbetaling.periode.fom + '-01', // TODO: 01 her er ein workaround i påvente av månadsveljar, vi bryr oss jo eigentleg ikkje om datoen her
+                tom: trekkILøpendeUtbetaling.periode.tom
+                    ? trekkILøpendeUtbetaling.periode.tom + '-01'
+                    : undefined,
+            },
+        };
+    }
+
+    const fjernFraLista = async (id: number) => {
+        settTrekkILøpendeUtbetalinger(trekkILøpendeUtbetalinger.filter(t => t.id !== id));
+    };
+
+    useEffect(() => {
+        hentTrekkILøpendeUtbetalinger();
+    }, [åpenBehandling]);
+
+    return !visTrekkILøpendeUtbetalinger && trekkILøpendeUtbetalinger.length === 0 ? (
+        <></>
+    ) : (
+        <>
+            <StyledHeading level="2" size="small" spacing>
+                Trekk i løpende utbetaling
+            </StyledHeading>
+            {trekkILøpendeUtbetalinger.map((trekkILøpendeUtbetaling: ITrekkILøpendeUtbetaling) => (
+                <TrekkILøpendeUtbetalingProvider
+                    key={trekkILøpendeUtbetaling.id}
+                    åpenBehandling={åpenBehandling}
+                    trekkILøpendeUtbetaling={trekkILøpendeUtbetaling}
+                    fjernFraLista={() => fjernFraLista(trekkILøpendeUtbetaling.id)}
+                    hentTrekkILøpendeUtbetalinger={hentTrekkILøpendeUtbetalinger}
+                >
+                    <TrekkILøpendeUtbetalingPanel key={trekkILøpendeUtbetaling.id} />
```

Comment:
Hvis man sender inn "er ny" til panelet her kan det brukes for å overstyre om det skal være ekspandert eller ikke

Reply:
💡